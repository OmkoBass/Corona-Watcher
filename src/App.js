import React, { useEffect, useState } from "react";

//Components
import Nav from "./Components/Nav";
import CoronaNumbers from "./Components/CoronaNumbers";
import Info from "./Components/Info";
import Countries from "./Components/Countries";
import Footer from "./Components/Footer";
import Loader from "./Components/Loader";
import GraphedCases from "./Components/GraphedCases";

import {
  getCountries,
  getGlobalData,
  getGlobalHistory,
} from "./Components/utils";

//css
import "./Styles/style.css";

//Bootstrap dependency
import "bootstrap/dist/css/bootstrap.min.css";

function App() {
  const [globalData, setGlobalData] = useState(null);
  const [countries, setCountries] = useState(null);
  const [globalHistory, setGlobalHistory] = useState(null);
  const [toTop, setToTop] = useState(false);
  const [done, setDone] = useState(false);

  async function setData() {
    getGlobalData().then((data) => setGlobalData(data));
    await getCountries().then((data) => setCountries(data));
    await getGlobalHistory().then((data) => setGlobalHistory(data));
  }

  function scroll() {
    if (
      document.body.scrollTop > 460 ||
      document.documentElement.scrollTop > 460
    ) {
      setToTop(true);
    } else {
      setToTop(false);
    }
  }

  window.onscroll = () => {
    scroll();
  };

  useEffect(() => {
    setData().then(function () {
      setDone(true);
    });
  }, []);

  return (
    <React.Fragment>
      <Nav />
      <div
        className="to-top"
        style={toTop ? null : { display: "none" }}
        onClick={() => window.scrollTo(0, 0)}
      >
        <span>&#8593;</span>
      </div>
      {done ? (
        <div>
          <CoronaNumbers
            update={globalData.updated}
            cases={globalData.cases}
            deaths={globalData.deaths}
            recovered={globalData.recovered}
          />
          <GraphedCases
            cases={globalHistory.cases}
            deaths={globalHistory.deaths}
            recovered={globalHistory.recovered}
          />
        </div>
      ) : (
        <Loader />
      )}
      <Info />
      {done ? <Countries countries={countries} /> : <Loader />}
      <Footer />
    </React.Fragment>
  );
}

export default App;
