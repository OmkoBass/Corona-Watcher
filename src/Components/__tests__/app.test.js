import React from "react";

import { render, screen, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

import App from "../../App";

afterEach(() => {
  cleanup();
});

it("Should render App", () => {
  render(<App />);

  const appElement = screen.getByTestId("app");

  expect(appElement).toBeInTheDocument();
});

it("Should render correctly", () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
