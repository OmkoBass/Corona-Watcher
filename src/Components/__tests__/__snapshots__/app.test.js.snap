// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  data-testid="app"
>
  <nav
    className="text-center text-white p-3"
    data-testid="nav"
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <h3>
      Corona Watcher
    </h3>
  </nav>
  <div
    className="to-top"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <span>
      ↑
    </span>
  </div>
  <div
    className="loading"
    data-testid="loader"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
  <div
    className="covid"
  >
    <h3>
      COVID-19
    </h3>
    <div
      className="container"
    >
      <h5>
        Coronavirus disease 2019 (COVID-19) (also known as Wuhan Novel Coronavirus) is an infectious disease caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), a virus closely related to the SARS virus. The disease first emerged in Wuhan, China and is the cause of the 2019–20 coronavirus outbreak. Those affected are likely to develop a fever and dry cough. Fatigue and shortness of breath are also common indicators. Sputum production, headache, muscle pain, and sore throat are less common symptoms.
      </h5>
      <p>
        (This information is pulled from covid-19 sources (link down below). I am not responsible if the info is not exactly right or it falls behind.
      </p>
      <hr />
    </div>
  </div>
  <div
    className="loading"
    data-testid="loader"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
  <footer
    className="p-3"
    data-testid="footer"
  >
    <a
      href="https://github.com/NovelCOVID/API"
    >
      https://github.com/NovelCOVID/API
    </a>
    <p>
      OmkoBass Github
    </p>
  </footer>
</div>
`;
